src/averageAges.js

```diff
@@ -37,7 +44,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people
+    .filter(({ name, sex }) => withChildren
+      ? people
+        .some(({ mother }) => mother === name)
```

Comment:
It is a smart solution.
But it will fail if we are going to have some man with the same name - theoretically :)
You still have to have `sex === 'f'` filter in place.